src/containers/Notifications.js

```diff
@@ -0,0 +1,151 @@
+import React from 'react'
+import { connect } from 'react-redux'
+import { prefetch } from 'react-fetcher'
+import { fetchActivity, markActivityRead, markAllActivitiesRead, navigate, thank } from '../actions'
+import { values, sortBy, isEmpty, contains } from 'lodash'
+import cx from 'classnames'
+import ScrollListener from '../components/ScrollListener'
+import A from '../components/A'
+import truncate from 'html-truncate'
+import { present, humanDate } from '../util/text'
+const { array, bool, func, number, object } = React.PropTypes
+
+@prefetch(({ dispatch }) => dispatch(fetchActivity(20, 0)))
+@connect(state => ({
+  activities: sortBy(values(state.activities), ['created_at']).reverse(),
```

Comment:
actually, this was because I was storing the activities as an object with their ids as keys. I rewrote to store them in an array. Not sure which I like better, but see https://github.com/Hylozoic/hylo-redux/commit/dbc5fe992094133b3d678bf1f57e3e898d8a0e7b for the array version

seems like the pagination fix is a bigger update. Let me know if you want me to have a crack at it.